From 15d09f55a687f5e5f26c5a5c7d9cabc36ee19512 Mon Sep 17 00:00:00 2001
From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Wed, 23 Jun 2010 12:15:13 +0200
Subject: [PATCH 1/2] [gstavidemux] reset the have_index flag at transition PAUSED_TO_READY

If we restart the Stream in the case of doing a transition from
PAUSED_TO_READY and back with READY_TO_PAUSED aso. the duration of the video
will get calculated even if we have a avi header with that information.

Signed-off-by: Michael Grzeschik <m.grzeschik@pengutronix.de>
---
 gst/avi/gstavidemux.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Index: gst-plugins-good-0.10.23/gst/avi/gstavidemux.c
===================================================================
--- gst-plugins-good-0.10.23.orig/gst/avi/gstavidemux.c	2010-06-23 17:36:48.000000000 +0200
+++ gst-plugins-good-0.10.23/gst/avi/gstavidemux.c	2010-06-23 17:36:50.000000000 +0200
@@ -5345,6 +5345,7 @@
 
   switch (transition) {
     case GST_STATE_CHANGE_PAUSED_TO_READY:
+      avi->have_index = FALSE;
       gst_avi_demux_reset (avi);
       break;
     default:
